indexing-service/src/main/java/org/apache/druid/indexing/overlord/ForkingTaskRunner.java

```diff
@@ -650,36 +653,38 @@ String getMaskedCommand(List<String> maskedProperties, List<String> command)
   }
 
   @Override
-  public long getTotalTaskSlotCount()
+  public Map<String, Long> getTotalTaskSlotCount()
   {
     if (config.getPorts() != null && !config.getPorts().isEmpty()) {
-      return config.getPorts().size();
+      return ImmutableMap.of(workerConfig.getCategory(), Long.valueOf(config.getPorts().size()));
     }
-    return config.getEndPort() - config.getStartPort() + 1;
+    return ImmutableMap.of(workerConfig.getCategory(), Long.valueOf(config.getEndPort() - config.getStartPort() + 1));
   }
 
   @Override
-  public long getIdleTaskSlotCount()
+  public Map<String, Long> getIdleTaskSlotCount()
   {
-    return Math.max(getTotalTaskSlotCount() - getUsedTaskSlotCount(), 0);
+    Map<String, Long> totalTaskSlots = getTotalTaskSlotCount();
+    Map<String, Long> usedTaskSlots = getUsedTaskSlotCount();
+    return ImmutableMap.of(workerConfig.getCategory(), Math.max(totalTaskSlots.get(workerConfig.getCategory()) - usedTaskSlots.get(workerConfig.getCategory()), 0));
```

Comment:
since I can't change return value of `getTotalTaskSlotCount()` method, are you saying that I shall add another method like `public long getTotalTaskSlotCountLong()`

Reply:
Yes, that is what I meant.